pkg/scheduler/scheduler.go

```diff
@@ -182,15 +185,16 @@ func (s *Scheduler) nominate(ctx context.Context, workloads []workload.Info, sna
 		ns := corev1.Namespace{}
 		e := entry{Info: w}
 		if snap.InactiveClusterQueueSets.Has(w.ClusterQueue) {
-			e.inadmissibleReason = fmt.Sprintf("ClusterQueue %s is inactive", w.ClusterQueue)
+			e.inadmissibleMsg = fmt.Sprintf("ClusterQueue %s is inactive", w.ClusterQueue)
 		} else if cq == nil {
-			e.inadmissibleReason = fmt.Sprintf("ClusterQueue %s not found", w.ClusterQueue)
+			e.inadmissibleMsg = fmt.Sprintf("ClusterQueue %s not found", w.ClusterQueue)
 		} else if err := s.client.Get(ctx, types.NamespacedName{Name: w.Obj.Namespace}, &ns); err != nil {
-			e.inadmissibleReason = fmt.Sprintf("Could not obtain workload namespace: %v", err)
+			e.inadmissibleMsg = fmt.Sprintf("Could not obtain workload namespace: %v", err)
 		} else if !cq.NamespaceSelector.Matches(labels.Set(ns.Labels)) {
-			e.inadmissibleReason = "Workload namespace doesn't match ClusterQueue selector"
+			e.inadmissibleMsg = "Workload namespace doesn't match ClusterQueue selector"
+			e.requeueReason = queue.RequeueReasonNamespaceMismatch
```

Comment:
We can't update in the workload controller without having the logic to prevent adding the workload to the queue on Add. As we discussed before, we can do that as followup since this is a long PR.

Reply:
I mean we should update the workload status to indicate that it was inadmissible for `namespaceSelector` not match.

Reply:
We are doing that, the inadmissibleMsg will show up in the workload status.

https://github.com/kubernetes-sigs/kueue/blob/73659ad78c0c935048914d94f94c756522c72133/pkg/scheduler/scheduler.go#L527

Reply:
Then it works well now, but if we put `namespaceSelector not match` workload to inadmissibleWorkloads in `PushOrUpdate`, then we'll not update the status at that moment. Just FYI.

Reply:
As I mentioned in the comment above, if we do that change, we need to update the workload controller as well to update the workload status.